src/Core/AuthenticationHelpers/AppServiceAuthentication.cs

```diff
@@ -69,11 +69,11 @@ public class AppServiceClaim
     {
         ClaimsIdentity? identity = null;
 
-        if (context.Request.Headers.TryGetValue(AuthenticationOptions.CLIENT_PRINCIPAL_HEADER, out StringValues header))
+        if (context.Request.Headers.TryGetValue(AuthenticationOptions.CLIENT_PRINCIPAL_HEADER, out StringValues headerValues) && headerValues.Count == 1)
```

Comment:
Should it be `out StringValues?` since the value may not exist?

Reply:
`out StringValues?` is a compiler error as the nullable type violates out param expected type
![image](https://github.com/Azure/data-api-builder/assets/6414189/7b347c2e-da50-409f-9c44-4628442470ac)

Reply:
if the value doesn't exist, and empty StringValues collection is returned per source code:
https://github.com/dotnet/aspnetcore/blob/8486d31e24f30e3fa1809a95699a0adc16f448d7/src/Http/Http.Features/src/IHeaderDictionary.cs#L14

```csharp
    /// <summary>
    /// IHeaderDictionary has a different indexer contract than IDictionary,
 where it will return StringValues.Empty for missing entries.
    /// </summary>
    /// <param name="key"></param>
    /// <returns>The stored value, or StringValues.Empty if the key is not present.</returns>
```

Reply:
this then begs the question: " so if an empty stringvalues collection is returned, how do you protect against processing an empty collection?"

- immediately after trygetvalue(), have to check that there is exactly one element in the collection.